### PR TITLE
[CIRC-1896] Logging - circulation resources: O - T

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Purpose
Cover from OverdueFineCirculationRulesEngineResource to TenantActivationResource by logging;

Resolves: [CIRC-1896](https://issues.folio.org/browse/CIRC-1896)

### CODE REVIEW

File: src/main/java/org/folio/circulation/resources/RequestByInstanceIdResource.java [hunk 1]
```diff
@@ -156,6 +158,8 @@ private CompletableFuture<Result<Map<Item, ZonedDateTime>>> getLoanItems(
     final List<Item> unsortedUnavailableItems = instanceRequestPackage.getUnsortedUnavailableItems();
     if (unsortedUnavailableItems == null ||
         unsortedUnavailableItems.isEmpty()) {
+      log.info("getLoanItems:: unsortedUnavailableItems is null or emtpy");
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/RequestByInstanceIdResource.java:161]:
```suggestion
      log.info("getLoanItems:: unsortedUnavailableItems is null or empty");
```

File: src/main/java/org/folio/circulation/resources/RequestHoldShelfClearanceResource.java [hunk 2]
```diff
@@ -173,27 +184,37 @@ private HoldShelfClearanceRequestContext createHoldShelfClearanceRequestContext(
...
+  private CompletableFuture<Result<HoldShelfClearanceRequestContext>>
+  findExpiredOrCancelledRequestByItemIds(GetManyRecordsClient client,
+    HoldShelfClearanceRequestContext context) {
+
+    List<Result<List<Request>>> requestList = findRequestsSortedByClosedDate(client,
+      context.getAwaitingPickupItemIds());
     List<Request> firstRequestFromList = getFirstRequestFromList(requestList);
-    return CompletableFuture.completedFuture(Result.succeeded(context.withExpiredOrCancelledRequests(firstRequestFromList)));
+    log.debug("findExpiredOrCancelledRequestByItemIds:: firstRequestFromList: {}",
+      () -> LogUtil.listAsString(firstRequestFromList));
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/RequestHoldShelfClearanceResource.java:195]: static import

File: src/main/java/org/folio/circulation/resources/RequestHoldShelfClearanceResource.java [hunk 3]
```diff
@@ -173,27 +184,37 @@ private HoldShelfClearanceRequestContext createHoldShelfClearanceRequestContext(
...
+  private Predicate<Request> hasNotContextAwaitingPickupRequestForItemId(
+    HoldShelfClearanceRequestContext context) {
+
     return req -> !context.getAwaitingPickupRequestItemIds().contains(req.getItemId());
   }
 
   private Result<List<Request>> findExpiredOrCancelledRequestByServicePoint(String servicePointId,
-                                                                            HoldShelfClearanceRequestContext context) {
+    HoldShelfClearanceRequestContext context) {
     List<Request> requestList = context.getExpiredOrCancelledRequests().stream()
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/RequestHoldShelfClearanceResource.java:213]:
```suggestion
    HoldShelfClearanceRequestContext context) {
    
    List<Request> requestList = context.getExpiredOrCancelledRequests().stream()
```

File: src/main/java/org/folio/circulation/resources/RequestNoticeSender.java [hunk 4]
```diff
@@ -190,6 +195,7 @@ private CompletableFuture<Result<Request>> fetchMissingLocationDetails(Request r
     Item item = request.getItem();
 
     if (item == null || item.isNotFound() || item.getLocation() == null) {
+      log.info("fetchMissingLocationDetails:: location cannot be fetched, item: {}", () -> item);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/RequestNoticeSender.java:198]: do we really need to log the whole item here in info? maybe an item ID will suffice?

**Review by OleksandrVidinieiev — APPROVED** [review]

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**